### PR TITLE
[shelly] Prepare for CrowdIn translations, bug fixes

### LINKED ISSUE #12535: [shelly] Binding shows log entries like "message.offline.status-error-watchdog"
Due to a definition problem in the language resources the binding shows messages like "message.offline.status-error-watchdog" instead of a message text

### LINKED ISSUE #12815: [shelly] endless initialization when incorrect credentials are used
The binding keeps initializing the thing when the configured credentials do not match the device settings. Immediately after state CONFIG_ERROR the binding switches back to INITIALIZING.

With the fix the Thing goes into state CONFIGURATION_ERROR: Access denied, check user id and password.
and stays OFFLINE

### PR DESCRIPTION
**Pull request by markus7017**
- Prepare for CrowsIn translation, shelly_de.properties deleted
- fix #12815 endless initialization; 
- fix #12535 invalid log messages

### CODE REVIEW

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/config/config.xml [hunk 1]
```diff
@@ -54,85 +53,84 @@
...
+			<label>@text/thing-type.config.shelly.eventsSwitch.label</label>
+			<description>@text/thing-type.config.shelly.eventsSwitch.description</description>
 			<advanced>true</advanced>
 			<default>false</default>
 		</parameter>
 		<parameter name="eventsPush" type="boolean" required="false">
-			<label>Enable Push Events (short/long)</label>
-			<description>True if the binding should register to get the Short/Long Push Events.</description>
+			<label>@text/thing-type.config.shelly.eventsSwitch.label</label>
+			<description>@text/thing-type.config.shelly.eventsSwitch.description</description>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/config/config.xml:133]: Mismatch: should be eventsPush

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/config/config.xml [hunk 2]
```diff
@@ -194,33 +191,33 @@
...
 			<default>true</default>
 		</parameter>
 		<parameter name="eventsCoIoT" type="boolean" required="false">
-			<label>Enable CoIoT Events</label>
-			<description>If enabled, the CoAP protocol is used to receive status updates.</description>
-			<advanced>true</advanced>
+			<label>@text/thing-type.config.shelly.eventsCoIoT.label</label>
+			<description>@text/thing-type.config.shelly.eventsCoIoT.description</description>
 			<default>true</default>
+			<advanced>true</advanced>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/config/config.xml:216]: Maybe the advanced property should be the same everywhere for this parameter ?
I see it sometimes true, sometimes false.

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/config/config.xml [hunk 3]
```diff
@@ -230,40 +227,40 @@
...
+			<label>@text/thing-type.config.shelly.battery.lowBattery.label</label>
+			<description>@text/thing-type.config.shelly.battery.lowBattery.description</description>
 			<default>20</default>
 			<unitLabel>%</unitLabel>
 		</parameter>
 		<parameter name="eventsSensorReport" type="boolean" required="false">
-			<label>Enable Sensor Events</label>
-			<description>True: Register event URL for sensor updates.</description>
+			<label>thing-type.config.shelly.eventsSensorReport.label</label>
+			<description>thing-type.config.shelly.eventsSensorReport.description</description>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/config/config.xml:251]: `@text/` is missing

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/config/config.xml [hunk 4]
```diff
@@ -272,34 +269,34 @@
...
-			<description>Password for HTTP API access.</description>
+			<label>@text/thing-type.config.shelly.password.label</label>
+			<description>@text/thing-type.config.shelly.password.description</description>
 			<context>password</context>
 		</parameter>
 		<parameter name="eventsSensorReport" type="boolean" required="false">
-			<label>Enable Sensor Events</label>
-			<description>True: Register event URL for sensor updates.</description>
+			<label>thing-type.config.shelly.eventsSensorReport.label</label>
+			<description>thing-type.config.shelly.eventsSensorReport.description</description>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/config/config.xml:287]: `@text/` is missing

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/thing/device.xml [hunk 5]
```diff
@@ -31,6 +31,7 @@
 				<option value="LOW_BATTERY">@text/channel-type.shelly.alarmTrigger.option.LOW_BATTERY</option>
 				<option value="BATTERY">@text/channel-type.shelly.alarmTrigger.option.BATTERY</option>
 				<option value="POWERON">@text/channel-type.shelly.alarmTrigger.option.POWERON</option>
+				<option value="EXT_POWER">@text/channel-type.shelly.alarmTrigger.option.EXT_POWER</option>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/thing/device.xml:34]: This entry is missing in the properties.

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/thing/relay.xml [hunk 6]
```diff
@@ -161,28 +161,28 @@
...
-				<label>@text/thing-type.shelly.shelly4pro.group.relay1.label</label>
+				<label>@text/channel-group-type.shelly.relayChannel1.label</label>
 			</channel-group>
 			<channel-group id="meter1" typeId="meter">
-				<label>@text/thing-type.shelly.shelly4pro.group.meter1.label</label>
+				<label>@text/channel-group-type.shelly.meter1.label</label>
 			</channel-group>
 			<channel-group id="relay2" typeId="relayChannel">
-				<label>@text/thing-type.shelly.shelly4pro.group.relay2.label</label>
+				<label>@text/channel-group-type.shelly.relayChannel1.label</label>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/thing/relay.xml:170]: Should be probably relayChannel2

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/thing/relay.xml [hunk 7]
```diff
@@ -161,28 +161,28 @@
...
 			</channel-group>
 			<channel-group id="meter2" typeId="meter">
-				<label>@text/thing-type.shelly.shelly4pro.group.meter2.label</label>
+				<label>@text/channel-group-type.shelly.meter2.label</label>
 			</channel-group>
 			<channel-group id="relay3" typeId="relayChannel">
-				<label>@text/thing-type.shelly.shelly4pro.group.relay3.label</label>
+				<label>@text/channel-group-type.shelly.relayChannel1.label</label>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/thing/relay.xml:176]: Should be probably relayChannel3

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/thing/relay.xml [hunk 8]
```diff
@@ -161,28 +161,28 @@
...
 			</channel-group>
 			<channel-group id="meter3" typeId="meter">
-				<label>@text/thing-type.shelly.shelly4pro.group.meter3.label</label>
+				<label>@text/channel-group-type.shelly.meter3.label</label>
 			</channel-group>
 			<channel-group id="relay4" typeId="relayChannel">
-				<label>@text/thing-type.shelly.shelly4pro.group.relay4.label</label>
+				<label>@text/channel-group-type.shelly.relayChannel1.label</label>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/thing/relay.xml:182]: Should be probably relayChannel4

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by markus7017** [comment]
A quick merge would help to speed up 😜 the CrowdIn translation bringing back German and maybe an additional language if the community is supporting.

How to remove the current German translation from CrowdIn? Several labels have changed so it's better to redo the translation.

**Comment by lolodomo** [comment]
> A quick merge would help to speed up ...

I will merge as soon as you correct the bugs I identified during the review.
So it can be fast if you are fast :)

**Comment by lolodomo** [comment]
> How to remove the current German translation from CrowdIn? Several labels have changed so it's better to redo the translation.

Good question. There is certainly something to do in Crowdin.
Once this PR will be merged, someone could invalidate all remaining German translations.

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/config/config.xml (hunk 4 above)
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/config/config.xml:287]: fixed

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/thing/device.xml (hunk 5 above)
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/thing/device.xml:34]: fixed

**Comment by markus7017** [comment]
Changes are applied - good catches! Thank you

> Once this PR will be merged, someone could invalidate all remaining German translations.

Who is "someone"? :-) Let's check after the merge

**Comment by lolodomo** [comment]
> Who is "someone"? :-) Let's check after the merge

Me for example ;) I am now unvalidating all entries. At least we are sure that a new Crowdin PR will not be created.

**Review by lolodomo — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
@markus7017 : this is ready in Crowdin. 16 pages before, 10 pages now, so you made a big reduction, good. There is less than one page with entries without translations, all others are present (already translated) but to be validated. If you want to propose another translation, you can.

**Comment by markus7017** [comment]
@lolodomo Translation on CrowdIn is completed. Is there a way to export the shelly_de.properties to do a final check?

**Comment by lolodomo** [comment]
> @lolodomo Translation on CrowdIn is completed. Is there a way to export the shelly_de.properties to do a final check?

First, someone speaking German has to validate all the translations in Crowdin.
Then a PR will be automatically created in Github with the new content of the file shelly_de.properties and you will be able to comment if you find something strange.

**Comment by markus7017** [comment]
> > @lolodomo Translation on CrowdIn is completed. Is there a way to export the shelly_de.properties to do a final check?
> 
> First, someone speaking German has to validate all the translations in Crowdin. Then a PR will be automatically created in Github with the new content of the file shelly_de.properties and you will be able to comment if you find something strange.

How to trigger that person? I'm German myself

**Comment by lolodomo** [comment]
0% approved because you requested to un-approve all of them.
The Crowdin manager for German translations is certainly @cweitkamp . He has to validate your proposals.

**Comment by markus7017** [comment]
@cweitkamp I verified the generated shelly_de.properties, which looks good.
In addition I added German translations for all missing io.*, persistence.*, voice.* and transform.* bundles
Have fun :-)

**Comment by cweitkamp** [comment]
:+1:

Great. Thank you very much.

**Comment by markus7017** [comment]
> 👍
> 
> Great. Thank you very much.

Let me know if there are more. It takes some time, but no sooo hard

I notice that the merge brought in DE translations like
```
channel-group-type.shelly.meter1.label = StromzÃ¤hler 1
channel-group-type.shelly.meter2.label = StromzÃ¤hler 2
channel-group-type.shelly.meter3.label = StromzÃ¤hler 3
channel-group-type.shelly.meter4.label = StromzÃ¤hler 4
```

Doesn't Crowdin uses UTF-8?

**Pull request merged**